rustler/src/types/atom.rs

```diff
@@ -197,6 +194,44 @@ unsafe impl Send for Atom {}
 /// The only overhead is checking that the atoms have been created (an atomic integer read).
 ///
 #[macro_export]
+macro_rules! atoms {
+    {
+        $(
+            $( #[$attr:meta] )*
+            $name:ident $( = $str:expr )*,
+        )*
+    } => {
+        #[allow(non_snake_case)]
+        struct RustlerAtoms {
+            $( $name : $crate::types::atom::Atom ),*
+        }
+        $crate::lazy_static::lazy_static! {
+            static ref RUSTLER_ATOMS: RustlerAtoms = $crate::env::OwnedEnv::new().run(|env| {
+                RustlerAtoms {
+                    $( $name: $crate::atoms!(@internal_make_atom(env, $name $( = $str)* )) ),*
+                }
+            });
+        }
+        $(
+            $( #[$attr] )*
+            pub fn $name() -> $crate::types::atom::Atom {
+                RUSTLER_ATOMS.$name
+            }
+        )*
+    };
+
+    // Internal helper macros.
+    { @internal_make_atom($env:ident, $name:ident) } => {
+        $crate::atoms!(@internal_make_atom($env, $name = stringify!($name)))
+    };
+    { @internal_make_atom($env:ident, $name:ident = $str:expr) } => {
+        $crate::types::atom::Atom::from_str($env, $str)
+            .ok().expect("rustler::atoms!: bad atom string")
```

Comment:
I think expect() should be enough here. Atom::from_str() returns NifResult<T>, which is just Result<T, Error>, and Result implements expect().

Reply:
See the corresponding [clippy lint](https://rust-lang.github.io/rust-clippy/master/#ok_expect). I think clippy cannot find this here as it is hidden in a macro.

Reply:
Yeah, I think I saw clippy complain about that in another PR...not sure why it's not consistently erroring on this for everything?

Reply:
I guess it cannot determine in this case that `Atom::from_str()` returns a NifResult because this is part of a macro and prefixed with `$crate`. I actually don't know if clippy can check macro_rules code at all.

Reply:
It looks like we can't change this here because `Error` doesn't implement `Debug`. We can make that change in #242 

```
error[E0599]: no method named `expect` found for type `std::result::Result<types::atom::Atom, error::Error>` in the current scope
   --> rustler/src/types/atom.rs:229:14
    |
229 |               .expect("rustler::atoms!: bad atom string")
    |                ^^^^^^
...
271 | / atoms! {
272 | |     /// The `nil` atom.
273 | |     nil,
274 | |
...   |
304 | |     last,
305 | | }
    | |_- in this macro invocation
    |
    = note: the method `expect` exists but the following trait bounds were not satisfied:
            `error::Error : std::fmt::Debug`
```

Reply:
:+1:

Reply:
@scrogson Did you mean #248?

Reply:
Yep, that one 😄